src/logger.ts

```diff
@@ -1,7 +1,12 @@
+import type {WriteStream} from 'fs';
 import assert from 'assert';
 import chalk from 'chalk';
 
-const LEVEL = Object.freeze({
+export type LogFormatter = (data: LogEvent) => string;
+export interface LogContext { requestId?: string }
+export interface LogEvent { level: string, time: string, requestId?: string, msg: string }
+
+const LEVEL: Record<string, number> = Object.freeze({
```

Comment:
Is the type really not inferred here? How odd.

Reply:
This happens when it's not defined:
![image](https://user-images.githubusercontent.com/5526/122685327-ea756480-d20a-11eb-8126-4978ad6113fb.png)